### DISCUSSION TITLE
wire:navigate in swiper not work

### DISCUSSION THREAD

**Discussion by rezahmady** [livewire/livewire#7410] [1. Help]
in swiper slider wire:navigate not work

**Comment by emildayan** [comment]
Any update here? Same issue.

**Comment by WilliamPinto-Olmos** [comment]
Hey @rezahmady, @emildayan. That could be because these anchor tags were added programatically after livewire initializes the wire:navigate directive or maybe because your swiper is overriding the wire:navigate action.

You could try using `Alpine.bind(link, { ["x-navigate"]: true })` to bind the x-navigate (or wire:navigate) functionality to a custom <a></a> tag. 

For example

```js
const link = document.createElement('a');

link.innerText = 'Im a link';
link.setAttribute('href', '/');
                
Alpine.bind(link, { ["x-navigate"]: true })

document.querySelector('.some-class').insertAdjacentElement('afterend', link);
```

The previous example binds programatically wire:navigate functionality to an anchor tag which doesnt even have wire:navigate attribute. So you could try binding the directive after the swiper is initialized.

**Comment by martinsnajdr** [comment]
I had this problem too. In my case it was because of the `allowTouchMove` + `touchStartPreventDefault` settings (default value for both is true). When you disable `touchStartPreventDefault`, the wire navigate should start working, but of course it will navigate to the url everytime you touch the slider even if you just want to swipe to next slide.

My (not elegant) solution was to add `locked` property inside my alpine component that inits the swiper, set `locked` to true when swiper `touchMove` event is fired, set `locked` to false when swiper `touchEnd` event is fired. So the `locked` is true everytime you touch move the slider.

Then I added a `navigate` method inside that component that checks if the `locked` property is true and if not, use Livewire.navigate to navigate to an url. I use this navigate method on the element that should act as anchor.

Do not forget to set `touchStartPreventDefault` to false.

**Comment by tuseto** [comment]
My solution is the following
Wrap swiper element in a div and set:
`x-on:click="navigateToUrl('/route/navigate/to')"`

And the method

`window.navigateToUrl = function (url) {
    Livewire.navigate(url);
}`

**Comment by the6fallenangel** [comment]
just do this:

`@click="Livewire.navigate('{{route()}}')"`
